src/core/common/debug.hpp

```diff
@@ -86,9 +86,11 @@
 
 #endif // OPENTHREAD_CONFIG_PLATFORM_ASSERT_MANAGEMENT
 
-#else
+#else // OPENTHREAD_CONFIG_ASSERT_ENABLE
 
-#define OT_ASSERT(cond)
+#include "openthread/platform/toolchain.h"
+
+#define OT_ASSERT(cond) OT_UNUSED_VARIABLE(cond)
```

Comment:
This won't work for all toolchains. Note that `OT_UNUSED_VARIABLE()` has different implemenations per toolchain limitation.
For example:
https://github.com/openthread/openthread/blob/d65f8a3812611a89f851a18df2d4b8219a6fbca2/include/openthread/platform/toolchain.h#L208-L214

The `cond` is a not necessarily a variable but a condition (statement) and the above logic won't work.

--------

Also this change seems to be unrelated to this PR and adding  SPI build check to CI. I would suggest to submit the assert changes as a separate PR. Thanks.

Reply:
This change is here because the assertion at https://github.com/openthread/openthread/blob/d65f8a3812611a89f851a18df2d4b8219a6fbca2/src/ncp/ncp_spi.cpp#L279 is the only place using the variable `readLength`. Build fails because of the "set but not used" error in some cases.
I changed this to use `OT_UNUSED_VARIABLE` after the assertion.

Reply:
Yes. That's the better/correct way (we already use this approach in other parts (where a variable is only used in an `OT_ASSERT()`))